File: api/src/modules/whatsapp-log/domain/entities/whatsapp-log.entity.ts
export type IWhatsappLog = {
  id: string;
  received_at: Date;

  // Device & routing
  device: string;
  mode: string;

  // Sender info
  sender: string;
  sender_lid: string | null;
  sender_name: string | null;

  // Group info
  is_group: boolean;
  group_id: string | null;
  member_phone: string | null;
  member_lid: string | null;

  // Message
  message_text: string | null;
  message_type: string;
  is_forwarded: boolean;
  is_quick: boolean;
  inbox_id: number;

  // Media
  extension: string | null;
  filename: string | null;
  url: string | null;
  location: string | null;

  // Poll
  poll_name: string | null;
  poll_choices: unknown | null;

  wa_timestamp: string;
};

File: handlerclaw/lib/features/whatsapp-logs/domain/entities/whatsapp_log_entity.dart
class WhatsappLogEntity {
  final String id;
  final String? userId;
  final DateTime receivedAt;
  final String device;
  final String mode;
  final String sender;
  final String? senderLid;
  final String? senderName;
  final bool isGroup;
  final String? groupId;
  final String? memberPhone;
  final String? memberLid;
  final String? messageText;
  final String messageType;
  final bool isForwarded;
  final bool isQuick;
  final int inboxId;
  final String? extension;
  final String? filename;
  final String? url;
  final String? location;
  final String? pollName;
  final dynamic pollChoices;
  final String waTimestamp;

  WhatsappLogEntity({
    required this.id,
    this.userId,
    required this.receivedAt,
    required this.device,
    required this.mode,
    required this.sender,
    this.senderLid,
    this.senderName,
    required this.isGroup,
    this.groupId,
    this.memberPhone,
    this.memberLid,
    this.messageText,
    required this.messageType,
    required this.isForwarded,
    required this.isQuick,
    required this.inboxId,
    this.extension,
    this.filename,
    this.url,
    this.location,
    this.pollName,
    this.pollChoices,
    required this.waTimestamp,
  });
}
